lab_1/main.py

```diff
@@ -4,80 +4,107 @@
 """
 
 
-def tokenize(text: str) -> list or None:
-    """
-    Splits a text into tokens, converts the tokens into lowercase,
-    removes punctuation and other symbols from words
-    :param text: a text
-    :return: a list of lower-cased tokens without punctuation
-    """
-    pass
+def tokenize(text: str):
+    if isinstance(text, str):
+        text = text.lower()
+        text = text.replace("'", ' ')
+        n_text = ''
+        for i in text:
+            if i.isalpha() or i == ' ' or i == '\n':
+                n_text += i
+            else:
+                text.replace(i, ' ')
+        t_tokens = n_text.split()
+        return t_tokens
+    return None
 
 
 def remove_stop_words(tokens: list, stop_words: list) -> list or None:
-    """
-    Removes stop words
-    :param tokens: a list of tokens
-    :param stop_words: a list of stop words
-    :return: a list of tokens without stop words
-    """
-    pass
+    if isinstance(tokens, list) and isinstance(stop_words, list):
+        no_stop_words = []
+        for token in tokens:
+            if token not in stop_words:
+                no_stop_words.append(token)
+        return no_stop_words
+    return None
 
 
-def calculate_frequencies(tokens: list) -> dict or None:
-    """
-    Calculates frequencies of given tokens
-    :param tokens: a list of tokens
-    :return: a dictionary with frequencies
-    """
-    pass
 
+def calculate_frequencies(tokens: list) -> dict or None:
+    if isinstance(tokens, list):
```

Comment:
There should be a check for the input arguments at the top of the function - after the documentation string. Remove nesting here and extract the check into a separate `if` block earlier.